### ISSUE TITLE
Bug in append mode for files (objects) larger than 5 MB

### ISSUE THREAD

**Issue by ToddSmall** [dask/s3fs#65]
A colleague (Danielle Fisher) found a bug in s3fs when appending to files larger than 5 MB. At [line 734](https://github.com/dask/s3fs/blob/master/s3fs/core.py#L734) of core.py, a path argument is passed to the boto `upload_part_copy` function. If this path has a `s3://` prefix, then an append will fail with the following error:

> ClientError: An error occurred (NoSuchBucket) when calling the UploadPartCopy operation: The specified bucket does not exist

If the path does not have the `s3://` prefix, then the append works just fine.

This code snippet will trigger the bug:

```
import s3fs

path = 's3://bucket/prefix/key'

fs = s3fs.S3FileSystem()

with fs.open(path, 'wb') as f:
    f.write(b'a' * (10 * 2 ** 20))

with fs.open(path, 'ab') as f:
    f.write(b'b' * (10 * 2 ** 20))
```

**Comment by mrocklin** [comment]
Thanks for the bug report and test.  Fixed in #66

**Comment by ToddSmall** [comment]
Thanks @mrocklin!
